core/src/main/java/org/apache/iceberg/GenericManifestFile.java

```diff
@@ -349,6 +362,9 @@ public Object get(int i) {
         this.partitions = value == null ? null :
             ((List<PartitionFieldSummary>) value).toArray(new PartitionFieldSummary[0]);
         return;
+      case 14:
+        this.keyMetadata = ByteBuffers.toByteArray((ByteBuffer) value);
```

Comment:
I think you'll need to do null guarding here as well,
```
this.keyMetadata = value == null ? null : .....
```

Reply:
`ByteBuffers.toByteArray` is null safe, that's why I did not add the check.

Reply:
Excellent!